src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,119 @@
+package duke;
+
+import duke.task.*;
+
+/**
+ * Parser helps to parse user input and call the relevant commands
+ * related to modifying the task list.
+ */
+public class Parser {
+    private TaskList tasks;
+    private Ui ui;
+
+    private Storage storage;
+
+    /**
+     * Constructs a Parser object.
+     *
+     * @param tasks A TaskList class containing tasks.
+     */
+    public Parser(TaskList tasks) {
+
+        this.tasks = tasks;
+        this.ui = new Ui();
+        this.storage = new Storage();
+    }
+
+    /**
+     * Parses the user input and calls relevant functions that modify
+     * the TaskList based on input.
+     *
+     * @param command User's input in a String Format.
+     */
+    public String parse(String command) throws DukeException{
+        String[] spacedCommand = command.split(" ");
+        String mainCommand = spacedCommand[0];
+        Task task;
+        String output = null;
+        try {
+            switch (mainCommand) {
```

Comment:
I like the indentations of this switch-case block that do not make it look like an arrowhead!